includes/components/class-woocommerce.php

```diff
@@ -165,9 +165,22 @@ public function get_related_product( $parent_id, $args = [] ) {
 	 * Formats price.
 	 *
 	 * @param float $price Price.
+	 * @param int   $product_id Product ID.
 	 * @return string
 	 */
-	public function format_price( $price ) {
+	public function format_price( $price, $product_id = null ) {
```

Comment:
Please move this logic below to the get_display_value, let's leave the helper simple, also the reason is that parameters are basically using the OR condition (e.g. if you provide the second one then the first one is not used at all)